test/refda/test_eval.c

```diff
@@ -457,3 +457,128 @@ void test_refda_eval_target_failure(const char *targethex, int expect_res)
+//
+// --- Division of TD value with primitive (int or float) ---
+// ari:/AC/(/TD/7000,/INT/1,//1/1/OPER/divide) -> /TD/7000
+TEST_CASE("821183820D191B588204018401012566646976696465", "820D191B58")
+// ari:/AC/(/TD/49000,/INT/7000,//1/1/OPER/divide) -> /TD/7000
+TEST_CASE("821183820D19BF688204191B588401012566646976696465", "820D191B58")
+// ari:/AC/(/TD/7000,/REAL32/3.5,//1/1/OPER/divide) -> /TD/2000
+TEST_CASE("821183820D191B588208F943008401012566646976696465", "820D1907D0")
+// ari:/AC/(/TD/7000,/REAL32/0.0,//1/1/OPER/divide) -> undefined
+TEST_CASE("821183820D191B588208F900008401012566646976696465", "F7")                            // Undefined Behavoir
+// ari:/AC/(/TD/7000,/REAL32/-INFINITY,//1/1/OPER/divide) -> /TD/0
+TEST_CASE("821183820D191B588208F9FC008401012566646976696465", "820D00")
+
+// Test various undefined behavoir involving TD and/or TP values
+//
+// --- Multiplication of primitive (int or float) with a TD value ---
+// ari:/AC/(/INT/1,/TD/7000,//1/1/OPER/multiply) -> undefined
+TEST_CASE("821183820401820D191B5884010125686D756C7469706C79", "F7")
+// ari:/AC/(/REAL32/3.5,/TD/7000,//1/1/OPER/multiply) -> undefined
+TEST_CASE("8211838208F94300820D191B5884010125686D756C7469706C79", "F7")
+//
+// --- Division of primitive (int or float) with a TD value ---
+// ari:/AC/(/INT/1,/TD/7000,//1/1/OPER/divide) -> undefined
+TEST_CASE("821183820401820D191B588401012566646976696465", "F7")
+// ari:/AC/(/REAL32/3.5,/TD/7000,//1/1/OPER/divide) -> undefined
+TEST_CASE("8211838208F94300820D191B588401012566646976696465", "F7")
+//
+// --- Addition,Subtraction,Multiplication,Division of TP value with a primitive (int) (in either order) ---
+// ari:/AC/(/TP/20000101T001640Z,/INT/1000,//1/1/OPER/add) -> undefined
+TEST_CASE("821183820C1903E882041903E88401012563616464", "F7")
+// ari:/AC/(/INT/1000,/TP/20000101T001640Z,//1/1/OPER/add) -> undefined
+TEST_CASE("82118382041903E8820C1903E88401012563616464", "F7")
+// ari:/AC/(/TP/20000101T001640Z,/INT/1000,//1/1/OPER/sub) -> undefined
+TEST_CASE("821183820C1903E882041903E88401012563737562", "F7")
+// ari:/AC/(/INT/1000,/TP/20000101T001640Z,//1/1/OPER/sub) -> undefined
+TEST_CASE("82118382041903E8820C1903E88401012563737562", "F7")
+// ari:/AC/(/TP/20000101T001640Z,/INT/1000,//1/1/OPER/multiply) -> undefined
+TEST_CASE("821183820C1903E882041903E884010125686D756C7469706C79", "F7")
+// ari:/AC/(/INT/1000,/TP/20000101T001640Z,//1/1/OPER/multiply) -> undefined
+TEST_CASE("82118382041903E8820C1903E884010125686D756C7469706C79", "F7")
+// ari:/AC/(/TP/20000101T001640Z,/INT/1000,//1/1/OPER/divide) -> undefined
+TEST_CASE("821183820C1903E882041903E88401012566646976696465", "F7")
+// ari:/AC/(/INT/1000,/TP/20000101T001640Z,//1/1/OPER/divide) -> undefined
+TEST_CASE("82118382041903E8820C1903E88401012566646976696465", "F7")
+void test_refda_eval_target_check(const char *targethex, const char *expectloghex)
+{
+    // TODO: This method is mostly duplicate from: test_refda_eval_target_valid()
```

Comment:
If this really is the same body the two can be combined. If the difference is the undefined result value, that can simply be handled by the expected value and that test assertion removed.